rules/no-array-for-each.js

```diff
@@ -314,14 +337,15 @@ function isFixable(callExpression, {scope, functionInfo, allIdentifiers, context
 	}
 
 	// Check `CallExpression.parent`
-	if (callExpression.parent.type !== 'ExpressionStatement') {
+	if (callExpression.parent.type !== 'ExpressionStatement' && callExpression.parent.type !== 'ChainExpression') {
 		return false;
 	}
 
-	// Check `CallExpression.callee`
-	// Because of `ChainExpression` wrapper, `foo?.forEach()` is already failed on previous check keep this just for safety
-	/* c8 ignore next 3 */
-	if (callExpression.callee.optional) {
+	if (isOptionalChaining && callExpression.parent.parent.type !== 'ExpressionStatement') {
```

Comment:
This is not right, https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1772 should fix this.